### PR TITLE
EASY: factor implementations of range length into a mixin template

### PR DESCRIPTION
**Pull request by andralex**
Arguably this is closer to correctness because some of these ranges don't define `opDollar`.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: std/range/package.d [hunk 1]
```diff
@@ -11786,12 +11734,10 @@ public:
         {
             return (*_range).length;
         }
-
```
Comment (wilzbach) [inline std/range/package.d:11789]: Unrelated!

Comment (andralex) [reply std/range/package.d:11789]: cool

Comment (andralex) [reply std/range/package.d:11789]: ah I noticed it was merged... guess it'll stay, yay
